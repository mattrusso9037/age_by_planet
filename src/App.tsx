import React from 'react';
import './App.css';
import {PlanetDisplay} from './components/Planets/PlanetDisplay';
import {ThemeProvider} from 'styled-components';
import styled from 'styled-components';
import { Header } from 'semantic-ui-react'

const theme = {
    fg: 'palevioletred',
    bg: 'white',
    baseUnit: 8,
};

const AppWrapper = styled.div`
background-image: url(./assets/space_bckg.webp);
background-position-x: center;
background-position-y: center;
background-repeat: no-repeat;
.description_container {
    width: 80%; 
    padding: 24px;
    font-size: 20px;
    line-height: 30px !important;
`;

function App() {
    return (
        <div className='App'>
            <AppWrapper>
                <ThemeProvider theme={theme}>
                    <header className='App-header'>
                        <Header style={{fontSize: 48}} as={'h1'}>Age By Planet</Header>
                        <div className={'description_container'}>
                            It takes the Earth about 365 days to orbit the sun, completing 1 year. How long is one year
                            on other planets, and how old would you be if you were living there?
                            Enter your age in the field below to determine your approximate age on each planet in our solar system!
                        </div>
                        <PlanetDisplay title={''}/>
                    </header>
                </ThemeProvider>
            </AppWrapper>
        </div>
    );
}

export default App;
