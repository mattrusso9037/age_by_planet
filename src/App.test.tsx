import React from 'react';
import {render, screen} from '@testing-library/react';
import App from './App';
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";

describe('App Component', () => {
    test('Renders Title', () => {
        render(<App/>);
        const linkElement = screen.getByText(/Age By Planet/i);
        expect(linkElement).toBeInTheDocument();
    });

    it("renders correctly", () => {
        const tree = shallow(<App/>);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('renders an image', () => {
       const wrapper = shallow(<App />);

       expect(wrapper.find('img').props().src).toEqual('pokeball_PNG8.png')
    });
});
