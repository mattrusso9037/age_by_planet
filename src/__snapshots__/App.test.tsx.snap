// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<div
  className="App"
>
  <styled.div>
    <Component
      theme={
        Object {
          "baseUnit": 8,
          "bg": "white",
          "fg": "palevioletred",
        }
      }
    >
      <header
        className="App-header"
      >
        <Header
          as="h1"
          className="planet_title"
        >
          Age By Planet
        </Header>
        <div
          className="description_container"
        >
          It takes the Earth about 365 days to orbit the sun, completing 1 year. How long is one year on other planets, and how old would you be if you were living there? Enter your age in the field below to determine your approximate age on each planet in our solar system!
        </div>
        <PlanetDisplay
          title=""
        />
      </header>
    </Component>
  </styled.div>
</div>
`;
